source/common/network/dns_resolver/dns_factory.cc

```diff
@@ -0,0 +1,59 @@
+#include "source/common/network/dns_resolver/dns_factory.h"
+
+namespace Envoy {
+namespace Network {
+
+// Create an empty c-ares DNS resolver typed config.
+void makeEmptyCaresDnsResolverConfig(
```

Comment:
Can you explain why passing by reference is preferred here?
It seems to me that returning the `typed_dns_resolver_config` is the preferred way, but maybe I'm missing something.

Reply:
I am passing the result as reference to avoid that copy when return value back to the caller.  But I think either way is fine.  Could you please let me know why return value is prefered way?